### PR TITLE
23424: fixed search with 0

### PR DESCRIPTION
**Pull request by dmytrokaplin**
### Description (*)
Fixed search with 0

### Fixed Issues (if relevant)
1. #23424 
2. https://github.com/magento/adobe-stock-integration/issues/100

### Manual testing scenarios (*)
1. Go to admin panel
2. Go to Catalog -> Products
3. Search products by 0

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=23427) <br/>All committers have signed the CLA.

File: app/code/Magento/CatalogSearch/Ui/DataProvider/Product/AddFulltextFilterToCollection.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ public function __construct(SearchCollection $searchCollection)
     public function addFilter(Collection $collection, $field, $condition = null)
     {
         /** @var $collection \Magento\Catalog\Model\ResourceModel\Product\Collection */
-        if (isset($condition['fulltext']) && !empty($condition['fulltext'])) {
+        if (isset($condition['fulltext']) && 0 !== strlen($condition['fulltext'])) {
```
Comment (VladimirZaets) [inline app/code/Magento/CatalogSearch/Ui/DataProvider/Product/AddFulltextFilterToCollection.php]: Hi, @jeysmook, thanks for collaboration. Please change the condition to `(string)$condition['fulltext'] !== ''`.

File: app/code/Magento/Ui/Component/Filters/Type/Search.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ protected function applyFilter()
     {
         $value = $this->getContext()->getRequestParam('search');
 
-        if ($value) {
+        if (0 !== strlen($value)) {
```
Comment (VladimirZaets) [inline app/code/Magento/Ui/Component/Filters/Type/Search.php]: The same here

**Review by VladimirZaets — CHANGES_REQUESTED** [review]

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-5402](https://jira.corp.magento.com/browse/ENGCOM-5402) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@jeysmook thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**